test/e2e/route_advertisements.go

```diff
@@ -222,3 +226,246 @@ var _ = ginkgo.Describe("BGP: Pod to external server when default podNetwork is
+				}
+			}
+
+			ginkgo.By("routes to the CUDN network are advertised to external frr router")
+			// Get the first element in the advertisements array (assuming you want to check the first one)
+			ginkgo.By("create route advertisement matching CUDN Network")
+			raManifest := generateRAManifest(testRAName)
+			cleanup, err := createManifest(f.Namespace.Name, raManifest)
+			ginkgo.DeferCleanup(cleanup)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("ensure route advertisement matching CUDN was created successfully")
+			gomega.Eventually(func() string {
+				podNetworkValue, err := e2ekubectl.RunKubectl("", "get", "ra", testRAName, "--template={{index .spec.advertisements 0}}")
+				if err != nil {
+					return ""
+				}
+				return podNetworkValue
+			}, 5*time.Second, time.Second).Should(gomega.Equal("PodNetwork"))
+
+			gomega.Eventually(func() string {
+				reason, err := e2ekubectl.RunKubectl("", "get", "ra", testRAName, "-o", "jsonpath={.status.conditions[?(@.type=='Accepted')].reason}")
+				if err != nil {
+					return ""
+				}
+				return reason
+			}, 30*time.Second, time.Second).Should(gomega.Equal("Accepted"))
+
+			ginkgo.By("queries to the external server are not SNATed (uses UDN podIP)")
+			podIP, err := podIPsForUserDefinedPrimaryNetwork(cs, f.Namespace.Name, clientPod.Name, namespacedName(f.Namespace.Name, testCudnName), 0)
+			framework.ExpectNoError(err, fmt.Sprintf("Getting podIPs for pod %s failed: %v", clientPod.Name, err))
+			framework.Logf("Client pod IP address=%s", podIP)
+			for _, serverContainerIP := range serverContainerIPs {
+				ginkgo.By(fmt.Sprintf("Sending request to node IP %s "+
+					"and expecting to receive the same payload", serverContainerIP))
+				cmd := fmt.Sprintf("curl --max-time 10 -g -q -s http://%s/clientip",
+					net.JoinHostPort(serverContainerIP, "8080"),
+				)
+				framework.Logf("Testing pod to external traffic with command %q", cmd)
+				stdout, err := e2epodoutput.RunHostCmdWithRetries(
+					clientPod.Namespace,
+					clientPod.Name,
+					cmd,
+					framework.Poll,
+					60*time.Second)
+				framework.ExpectNoError(err, fmt.Sprintf("Testing pod to external traffic failed: %v", err))
+				if isIPv6Supported() && utilnet.IsIPv6String(serverContainerIP) {
+					podIP, err := podIPsForUserDefinedPrimaryNetwork(cs, f.Namespace.Name, clientPod.Name, namespacedName(f.Namespace.Name, testCudnName), 1)
+					framework.ExpectNoError(err, fmt.Sprintf("Getting podIPs for pod %s failed: %v", clientPod.Name, err))
+					// For IPv6 addresses, need to handle the brackets in the output
+					outputIP := strings.TrimPrefix(strings.Split(stdout, "]:")[0], "[")
+					gomega.Expect(outputIP).To(gomega.Equal(podIP),
+						fmt.Sprintf("Testing pod %s to external traffic failed while analysing output %v", echoClientPodName, stdout))
+				} else {
+					// Original IPv4 handling
+					gomega.Expect(strings.Split(stdout, ":")[0]).To(gomega.Equal(podIP),
```

Comment:
what if ipv4 isnt supported?

Reply:
standalone v6 isn't supported is my understanding :D